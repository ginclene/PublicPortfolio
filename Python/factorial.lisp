; This is a comment - your code should ignore these

; A recursive program
(define fact
  (lambda (n) ; Computes n!
     (if (< n 1)
     ; then
         1
     ; else
        (* n (fact (- n 1)))
     ); end if
   ); end lambda
); end fact

(fact 4)

; Problem 5.6 converted to scheme
(define x 0)
(define sub1 (lambda (x)
     ( ; this open paren is to _call_ sub2
         (lambda ()  ; this lambda function is sub2
          (sub3) ; this calls sub3
          ) ; this ends sub2
      ) ; this close paren ends the _call_ to sub2
     )  ;end the lambda for sub1
 ) ; end the define for sub1

(define sub3 (lambda () x)) ; returns x, whatever that is
(sub1 2)
