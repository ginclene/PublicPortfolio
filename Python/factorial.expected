GLOBAL
--------------------
fact      : Procedure
--------------------
n         : 4
--------------------
GLOBAL
--------------------
fact      : Procedure
--------------------
n         : 3
--------------------
GLOBAL
--------------------
fact      : Procedure
--------------------
n         : 2
--------------------
GLOBAL
--------------------
fact      : Procedure
--------------------
n         : 1
--------------------
GLOBAL
--------------------
fact      : Procedure
--------------------
n         : 0
--------------------
24
GLOBAL
--------------------
fact      : Procedure
x         : 0
sub1      : Procedure
sub3      : Procedure
--------------------
x         : 2
--------------------
GLOBAL
--------------------
fact      : Procedure
x         : 0
sub1      : Procedure
sub3      : Procedure
--------------------
x         : 2
--------------------
--------------------
GLOBAL
--------------------
fact      : Procedure
x         : 0
sub1      : Procedure
sub3      : Procedure
--------------------
--------------------
0
